EasyImport.py

```diff
@@ -366,196 +365,206 @@ def setCurrentConfig(self, idConfig):
+            newFeature: object = QgsFeature(fields)
             initFields = destinationlayer.dataProvider().fields()
             newFeature.setFields(initFields)
             newFeature.initAttributes(initFields.size())
             newFeature.setGeometry(QgsGeometry.fromWkt(str(feature.GetGeometryRef())))
-                        
+
             # For each column mapping
             for k in colunmMappingDict.keys():
                 # print "source %s -> destination %s" % (k,colunmMappingDict[k])
-                
+
                 # With regex
                 if regexMappingDict[k]:
                     regexp = re.compile(regexMappingDict[k])
                     # print str(k)
                     # print feature.GetField(str(k))
-                    
+
                     m = regexp.match(feature.GetField(str(k)))
                     if not (m is None):
                         # print 'Regex : %s   --> Value : %s' % (regexMappingDict[k], m.group('group'))
-                        newFeature.setAttribute(fields.fieldNameIndex(colunmMappingDict[k]), m.group('group'))
-                
+                        newFeature.setAttribute(fields.lookupField(colunmMappingDict[k]), m.group('group'))
+
                 # Without regex
                 else:
-                    newFeature.setAttribute(fields.fieldNameIndex(colunmMappingDict[k]), feature.GetField(str(k)))
-            
+                    newFeature.setAttribute(fields.lookupField(colunmMappingDict[k]), feature.GetField(str(k)))
+
             # For each static value mapping
             for k in staticMappingDict.keys():
-                #print "destination %s -> value %s" % (k,staticMappingDict[k])
-                newFeature.setAttribute(fields.fieldNameIndex(k), staticMappingDict[k])
-                
+                # print "destination %s -> value %s" % (k,staticMappingDict[k])
+                newFeature.setAttribute(fields.lookupField(k), staticMappingDict[k])
+
             # Get default value for oid
             # See https://github.com/qgis/QGIS/pull/5378
-            # When migrating to QGIS 3, remove following line 
+            # When migrating to QGIS 3, remove following line
             # and use QgsVectorLayerUtils.createFeature instead of layer.addFeature
-            if destinationlayer.fieldNameIndex('obj_id') >= 0:
-                newFeature['obj_id'] = destinationlayer.dataProvider().defaultValue(destinationlayer.fieldNameIndex('obj_id'))
-            
+            #if destinationlayer.fields().indexFromName('obj_id') >= 0:
+            #    newFeature['obj_id'] = destinationlayer.dataProvider().defaultValue(
+            #        destinationlayer.fields().indexFromName('obj_id'))
+
             # Add new feature in destination layer
-            destinationlayer.addFeature(newFeature, True)
-            
+            #destinationlayer.addFeature(newFeature, True) Removed
+            QgsVectorLayerUtils.createFeature(destinationlayer)
```

Comment:
Ping @3nids

Hello! 

Pourrais tu me faire une revue de code par ici?

J'ai bien compris qu'il manque qqch vu que la définition de QgsVectorLayerUtils.createFeature() spécifie "Note that the created feature is not automatically inserted into the layer." mais je ne comprend pas ce qu'il faut faire pour l'insérer, et je n'ai pas trouvé d'exemple illustrant mon besoin sur le net ni dans la doc pyqgis testing).

Bonne journée!

Reply:
Hello,
createFeature retourne un objet feature.
I'll faut donc faire newFeature = QgsVectorLayerUtils.createFeature(destinationLayer) mais le déplacer avant la définition des attributs ou de la géométrie.

Reply:
Avec plaisir !